mtags/src/main/scala/scala/meta/internal/pc/AutoImportsProvider.scala

```diff
@@ -0,0 +1,148 @@
+package scala.meta.internal.pc
+
+import scala.meta.pc.OffsetParams
+import org.eclipse.{lsp4j => l}
+import scala.meta.pc.SymbolSearchVisitor
+import java.nio.file.Path
+import scala.meta.pc.AutoImportsResult
+import scala.collection.mutable
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+final class AutoImportsProvider(
+    val compiler: MetalsGlobal,
+    name: String,
+    params: OffsetParams
+) {
+  import compiler._
+
+  def autoImports(): List[AutoImportsResult] = {
+    val unit = addCompilationUnit(
+      code = params.text,
+      filename = params.filename,
+      cursor = Some(params.offset)
+    )
+    val pos = unit.position(params.offset)
+    val importPosition = autoImportPosition(pos, params.text())
+    val context = doLocateImportContext(pos, importPosition)
+    val isSeen = mutable.Set.empty[String]
+    val symbols = List.newBuilder[Symbol]
+
+    def visit(sym: Symbol): Boolean = {
+      val id = sym.fullName
+      if (!isSeen(id)) {
+        isSeen += id
+        symbols += sym
+        true
+      }
+      false
+    }
+
+    val visitor = new CompilerSearchVisitor(name, context, visit)
+
+    search.search(name, buildTargetIdentifier, visitor)
+
+    symbols.result.collect {
+      case sym if sym.name.dropLocal.decoded == name =>
+        val ident = Identifier.backtickWrap(sym.name.dropLocal.decoded)
+        val pkg = sym.owner.fullName
+        val edits = importPosition match {
+          case None => Nil
+          case Some(value) =>
+            val (short, edits) = ShortenedNames.synthesize(
+              TypeRef(ThisType(sym.owner), sym, Nil),
+              pos,
+              context,
+              value
+            )
+            edits
+        }
+        AutoImportsResultImpl(pkg, edits.asJava)
+    }
+  }
+
+  private class CompilerSearchVisitor(
```

Comment:
this is analogous to the homonymous class in `CompletionProvider`, except we visit symbols directly (instead of normalizing everything into a completion `Member`)

Reply:
Might make sense to use the same logic with an additional transformer (TextEdit => T) function that might be identity by default.

Reply:
Done, take a look!